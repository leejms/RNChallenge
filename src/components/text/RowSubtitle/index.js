/**
 * RowSubtitle
 *
 * Text style for the secondary text on cells and rows
 * text content is based on children like the standard Text component
 */
//import liraries
import React, {Component} from 'react'
import {View, Text, StyleSheet} from 'react-native'

const Theme = require('../../../styles')

// create a component
const RowSubtitle = ({children}) => {
  return (
    <View style={styles.container}>
      <Text style={Theme.textStyles.subtitleText}>{children}</Text>
    </View>
  )
}

// define your styles
const styles = StyleSheet.create({container: {}})

//make this component available to the app
export default RowSubtitle
