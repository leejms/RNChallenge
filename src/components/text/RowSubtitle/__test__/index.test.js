import 'react-native'
import React from 'react'
import RowSubtitle from '../index'

import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const tree = renderer.create(
    <RowSubtitle>Test</RowSubtitle>
    ).toJSON()
  expect(tree).toMatchSnapshot()
})