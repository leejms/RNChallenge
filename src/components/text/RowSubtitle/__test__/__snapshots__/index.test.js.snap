// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={Object {}}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#a5a5a5",
        "fontFamily": "Roboto-Regular",
        "fontSize": 14,
      }
    }
  >
    Test
  </Text>
</View>
`;
